File: src/main/java/pw/switchcraft/plethora/integration/computercraft/meta/item/PrintoutItemMeta.java
package pw.switchcraft.plethora.integration.computercraft.meta.item;

import dan200.computercraft.shared.media.items.ItemPrintout;
import net.minecraft.item.ItemStack;
import pw.switchcraft.plethora.api.meta.ItemStackMetaProvider;

import javax.annotation.Nonnull;
import java.util.HashMap;
import java.util.Map;

public final class PrintoutItemMeta extends ItemStackMetaProvider<ItemPrintout> {
    public PrintoutItemMeta() {
        super("printout", ItemPrintout.class);
    }

    @Nonnull
    @Override
    public Map<String, ?> getMeta(@Nonnull ItemStack stack, @Nonnull ItemPrintout printout) {
        Map<String, Object> out = new HashMap<>(4);
        out.put("type", printout.getType().toString());
        out.put("title", ItemPrintout.getTitle(stack));
        out.put("pages", ItemPrintout.getPageCount(stack));

        Map<Integer, String> lines = new HashMap<>();
        String[] lineArray = ItemPrintout.getText(stack);
        for (int i = 0; i < lineArray.length; i++) {
            lines.put(i + 1, lineArray[i]);
        }
        out.put("lines", lines);

        return out;
    }
}

File: src/main/java/io/sc3/plethora/integration/computercraft/meta/item/PrintoutItemMeta.kt
package io.sc3.plethora.integration.computercraft.meta.item

import dan200.computercraft.shared.ModRegistry
import dan200.computercraft.shared.media.items.PrintoutItem
import io.sc3.plethora.api.meta.ItemStackMetaProvider
import net.minecraft.item.ItemStack

class PrintoutItemMeta : ItemStackMetaProvider<PrintoutItem>(PrintoutItem::class.java, "printout") {
  override fun getMeta(stack: ItemStack, item: PrintoutItem): Map<String, *> {
    val lines: MutableMap<Int, String> = HashMap()
    val lineArray = PrintoutItem.getText(stack)
    for (i in lineArray.indices) {
      lines[i + 1] = lineArray[i]
    }

    val type = when {
      item == ModRegistry.Items.PRINTED_BOOK.get() -> "book"
      item == ModRegistry.Items.PRINTED_PAGES.get() -> "pages"
      item == ModRegistry.Items.PRINTED_PAGE.get() -> "page"
      else -> "unknown"
    }

    return mapOf(
      "type"  to type,
      "title" to PrintoutItem.getTitle(stack),
      "pages" to PrintoutItem.getPageCount(stack),
      "lines" to lines
    )
  }
}
